temp/automatic_ticket_revocations_proposal.md

```diff
@@ -0,0 +1,123 @@
+
+- General Audience
+- Technical Details
+
+The general audience section provides a high-level overview of the proposal.  It includes details suitable for all stakeholders, including those unfamiliar with the inner workings of consensus.
+
+The technical details section contains a more in-depth discussion of the relevant technical aspects, implementation specifics, and rationale for the chosen semantics of the proposal.
+
+# General Audience
+
+## High-Level Overview
+
+This proposal intends to significantly improve the stakeholder user experience around ticket revocations by automatically revoking missed or expired tickets.
+
+Further, it will enable the recovery of funds for users who lost their redeem script for the legacy VSP system (before the release of [vspd](https://blog.decred.org/2020/06/02/A-More-Private-Way-to-Stake/), which removed the need for the redeem script).
+
+## Current State of Ticket Revocations
+
+Missed and expired tickets must be revoked to return the funds used to purchase the ticket to the user.  Missed and expired tickets are defined as follows:
+
+- **Missed tickets** are tickets that fail to cast a vote when selected.  Tickets are pseudorandomly selected to vote for each block.  If the wallet with voting rights for that ticket is not online to cast its vote when a ticket is selected to vote, the ticket is marked as missed.
+
+- **Expired tickets** are tickets that are not selected to vote before the ticket expiration duration (40,960 blocks, or approximately 142 days, on mainnet).
+
+Revoking a missed or expired ticket requires broadcasting a **ticket revocation transaction**.  In order to manually revoke a ticket today, a user needs to:
+- Open the wallet that has voting rights for the ticket so that the ticket revocation transaction can be signed
+- If the wallet is running in SPV mode, the user will need to disable SPV, sync the blockchain, and rescan
+- Revoke the missed or expired ticket(s)
+
+## Automatic Ticket Revocations
+
+This proposal proposes the following changes to enable automatic ticket revocations:
+- Allow tickets to be revoked by anyone instead of requiring a valid signature from the voting address
+	- Ticket revocations will still have to go to the original commitment address in the ticket, as they do today, so this does not allow theft of funds
+	- This would not apply to votes, and therefore voting rights for tickets would remain as they are today
+- Require blocks to contain ticket revocation transactions for all tickets that are missed or expired as of that block
+
+## Consensus Rule Change Cost-Benefit Analysis
+
+When considering changes to the consensus rules, it is paramount to examine both the positive and negative consequences of the proposed changes.  There are nearly always tradeoffs involved in one form or another.  Since there are no known agreed-upon cons to these changes, the following analysis contains one section for pros and another for an expected potential objection.
```

Comment:
There are some additional potential cons I can think of, depending on whose point of view you're considering it from:

- It removes fees from revocations which slightly reduces potential income to PoW miners
- All wallet implementation will need to be updated to accommodate the proposed changes which involves both back-end modifications as well as front-end design work
- All custom mining template generation outside of dcrd will also have to be updated to properly include the required revocations which might be a non-trivial amount of work

I would also suggest a bit of a recap after with a brief discussion addressing those cons, because, a couple of them are a matter of perspective.  e.g. stake hodlers not having to pay another transaction fee after already having their ticket expire would likely be welcomed, while PoW miners would prefer the additional fees.

Reply:
Thanks for outlining these!  I've updated this section to describe the various cons as you recommended.  I think I was originally not considering the work outside of `dcrd` a con and more of just required work that comes with accommodating any consensus change, but it makes sense and I agree that any additional required work should always be explicitly listed as a tradeoff to be considered.